File: array_problems/capitalise_vowels/capitalise_vowels.php
<?php
class ArrayProblems {
    public function capitalise_vowels($str) {
        $vowels = [ 'a', 'e', 'i', 'o', 'u' ];
        $res = "";
        for($i = 0; $i < strlen($str); $i++) {
            $currentChar = $str[$i];
            if(in_array($currentChar, $vowels)) {
                $uppercase = mb_strtoupper($currentChar);
                $res .= $uppercase;
            } else {
                $res .= $currentChar;
            }
        }
        return $res;
    }
}
$arrayProblems = new ArrayProblems();
$str = "We can also traverse the array elements using foreach loop. It returns array element one by one";
$capitalised_vowels = $arrayProblems->capitalise_vowels($str);
echo($capitalised_vowels);

File: array_problems/capitalise_vowels/capitalise_vowels.cs
using System;
using System.Threading.Tasks;
using System.Collections.Generic;
class ArrayProblems {
    public async Task<string> capitalise_vowels(string str) {
        List<char> vowels = new List<char> { 'a', 'e', 'i', 'o', 'u' };
        string res = "";
        for (int i = 0; i < str.Length; i++) {
            char currentChar = str[i];
            if(vowels.Contains(char.ToLower(currentChar))) {
                char uppercase = char.ToUpper(currentChar);
                res += uppercase;
            } else {
                char lowercase = char.ToLower(currentChar);
                res += lowercase;
            }
        }
        return res;
    }
}

public class Program {
    public static async Task Main(string[] args) {
        Console.WriteLine("Enter a string:");
        string str = Console.ReadLine();
        ArrayProblems arrayProblems = new ArrayProblems();
        string capitalised_vowels = await arrayProblems.capitalise_vowels(str);
        Console.WriteLine(capitalised_vowels);
    }
}